samples/Ark.ReferenceProject/Agents.md

```diff
@@ -13,6 +13,12 @@
 - Follow existing patterns - check `Ping` entity implementation as reference
 - Place DTOs in `*.API` project, handlers in `*.Application` project
 - Add Reqnroll BDD tests for new features
+- Use a single `JsonSerializerContext` for all types serialized by the application (Requests, Queries, Messages)
+- Configure `JsonSerializerContext` with Ark defaults using a helper method in the Application layer (e.g., `Ex.CreateCoreApiJsonSerializerOptions()`) instead of creating options inline
+- Register `JsonSerializerContext` using `TypeInfoResolver` pattern, not `TypeInfoResolverChain`
+- Note: `JsonSerializerOptions` get locked when passed to a `JsonSerializerContext` constructor, preventing reuse for multiple contexts - create separate instances for each context
+- **BusinessRuleViolations**: Derive from `Ark.Tools.Core.BusinessRuleViolation.BusinessRuleViolation`, specialize with domain-specific properties (e.g., `BookPrintingProcessAlreadyRunningViolation` with `BookId` property). The class name itself serves as the error code. See `samples/WebApplicationDemo/Dto/CustomBusinessRuleViolation.cs` as an example
```

Comment:
Remove references from WebApplicationDemo as the reference project is supposed to be a scaffold template and the Agents.md must remain valid when used in a new project